modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -173,21 +286,26 @@ public void processOpts() {
             throw new IllegalArgumentException("aspnetCoreVersion must be '2.1', '2.0' but found " + aspnetCoreVersion);
         }
 
+
         supportingFiles.add(new SupportingFile("build.sh.mustache", "", "build.sh"));
         supportingFiles.add(new SupportingFile("build.bat.mustache", "", "build.bat"));
         supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
         supportingFiles.add(new SupportingFile("Solution.mustache", "", packageName + ".sln"));
-        supportingFiles.add(new SupportingFile("Dockerfile.mustache", packageFolder, "Dockerfile"));
         supportingFiles.add(new SupportingFile("gitignore", packageFolder, ".gitignore"));
-        supportingFiles.add(new SupportingFile("appsettings.json", packageFolder, "appsettings.json"));
-
-        supportingFiles.add(new SupportingFile("Startup.mustache", packageFolder, "Startup.cs"));
-        supportingFiles.add(new SupportingFile("Program.mustache", packageFolder, "Program.cs"));
```

Comment:
I am wonder if the supporting files could be pulled behind a factory. It could switch off whether it is a library or a full solution. 

It would just unload a lot of lines/logic form this class and make it easier to read.

Reply:
Possibly but it might make the set of files that are generated harder to follow.

Reply:
In my mind, it might make it easier to following if I just wanted to edit the lib section. I could know in one go what is being edited.

Reply:
How about we do it in a second pass - the code is simple at this time and generally follows the pattern. I can see files being in and out based on some configuration in the long term and it should be table driven rather than in the code directly.

Reply:
I actually used this processed pattern (strategy) in another generator improvement. I only used it for `additionalProperties` as a way to avoid many control structures, but it could easily be applied to other templating properties as well.

I don't think redundancy in such types is a huge concern if it results in code that's easier to follow.

See https://github.com/OpenAPITools/openapi-generator/pull/2187 if you're interested in checking it out.